### DISCUSSION TITLE
Help understanding dependencies: `ilmbase` vs `imath` and `openexr`

### DISCUSSION THREAD

**Discussion by tjluoma** [Homebrew/discussions#1436] [Everyday usage] [answered]
At the risk of appearing dumber than I think I am, I have questions about two `brew doctor` warnings which I don't know what to do with, because I don't know what the related tools are used for.

First, let me show you two messages… first from `brew doctor`

### `brew doctor`
```
Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  ilmbase

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  imath
  openexr
```

### And then from `brew upgrade`

```
Warning: ilmbase has been deprecated because it is not supported upstream!
==> Upgrading 5 outdated packages:
mas 1.8.1 -> 1.8.2
ilmbase 2.5.5 -> 2.5.5_1
carthage 0.37.0 -> 0.38.0
openexr 2.5.5 -> 3.0.1
imagemagick 7.0.11-11 -> 7.0.11-11_1
==> Upgrading ilmbase 2.5.5 -> 2.5.5_1 
…
==> Pouring ilmbase--2.5.5_1.arm64_big_sur.bottle.tar.gz
==> Caveats
ilmbase is keg-only, which means it was not symlinked into /opt/homebrew,
because ilmbase conflicts with `openexr` and `imath`.
```

## Questions

1. I don't know what `ilmbase` or `imath` or `openexr` do, but I assume they are a "dependency" of another tool that I have installed. I don't know how to find out which tool. Is there a way to ask `brew` to tell me which tools rely on them?

2. If `ilmbase` is a dependency of another tool, there's not much I can do about finding a replacement, is there? That really depends on the developer of whichever tool requires it. I guess my only option is to not use the tool which requires it, but… back to question #1, I'm not sure how to figure that out.

3. It says `ilmbase` is deprecated and not supported, but there was also an update to `ilmbase` (`ilmbase 2.5.5 -> 2.5.5_1`) so I'm confused… doesn't that mean that someone is supporting it? Obviously not, but I'm not sure what it does mean.

**Comment by carlocab** [comment, marked as answer]
`openexr` used to depend on `ilmbase`, but `ilmbase` has since been replaced with `imath` (which is also why `ilmbase` is deprecated). You can check whether you have anything installed that still uses `ilmbase`:
```
brew uses --installed ilmbase
```
You can also link `openexr` and `imath` with
```
brew link openexr imath
```
They were probably not linked properly because `openexr` was upgraded and `imath` installed before `ilmbase` was "upgraded" (and hence unlinked).

The `ilmbase` update wasn't a real update: it was just a revision bump to make sure that it is unlinked to avoid conflicting with an `openexr`/`imath` install.

**Comment by tjluoma** [comment]
Thank you! That is a very helpful explanation.

I ran `brew uses --installed ilmbase` and nothing was returned. If I understand correctly, that means nothing is using it?

Thank you again for helping me understand `brew` a little better.

**Reply by carlocab** [reply]
You understand correctly. You should be good to remove it.

**Reply by tjluoma** [reply]
Thanks
